src/org/thoughtcrime/securesms/WelcomeActivity.java

```diff
@@ -96,6 +101,32 @@ private void registerForEvents() {
         eventCenter.addObserver(DcContext.DC_EVENT_IMEX_PROGRESS, this);
     }
 
+    private void handleIntent() {
+        if (getIntent() != null && Intent.ACTION_VIEW.equals(getIntent().getAction())) {
+            Uri uri = getIntent().getData();
+            if (uri == null) return;
+
+            if (uri.getScheme().equalsIgnoreCase(DCACCOUNT)) {
+                String uriString = uri.toString();
+                uriString = uriString.replaceFirst(DCACCOUNT, DCACCOUNT.toUpperCase());
```

Comment:
Is this line really necessary? Also, it seems to be the only difference to the `else if` block?

Reply:
simplified :heavy_check_mark: